qemu/tests/hugepage_specify_node.py

```diff
@@ -63,7 +66,16 @@ def run(test, params, env):
                                   idle_node, test.log.info)
         params["setup_hugepages"] = "yes"
         hp_config = test_setup.HugePageConfig(params)
-        hp_config.setup()
+        try:
+            hp_config.setup()
+        except ValueError as details:
+            # Check if there have enough contiguous Memory on the host node
+            details = str(details)
+            if re.findall(r"Cannot set %s hugepages" % huge_pages_num, details):
+                hp_config.cleanup()
```

Comment:
Because the system does not know whether its CMA (Contiguous Memory Allocator) is sufficient at the beginning of the allocation.
That is to say, it will allocate until there are not enough contiguous huge pages. 
Please see the example below:
```
]# echo 4 > /sys/devices/system/node/node0/hugepages/hugepages-1048576kB/nr_hugepages
]# cat /sys/devices/system/node/node0/hugepages/hugepages-1048576kB/nr_hugepages
2
```
This is why the framework compares the number of pages, because the system will set the page. It just stops when the CMA is insufficient, and the huge pages that have been allocated before will not be released automatically.
https://github.com/avocado-framework/avocado-vt/blob/master/virttest/test_setup/__init__.py#L531

Reply:
I got your point, but we always clean up at last the `finally:` block, right?
line 96~99:
```
        finally:
            vm.destroy()
            hp_config.cleanup()
            time.sleep(10)
```
From my understanding, both of them are doing the same work, just executed in different periods.
I think we could keep the `finally` one if you mentioned `the huge pages that have been allocated before will not be released automatically.`

Reply:
nice capture!
has been corrected.